### PR TITLE
Add `PointSpace`

### PR DESCRIPTION
**Pull request by jb-mackay**
Adds a 0D `PointSpace` to `ClimaCore.Spaces`. The `PointSpace` allows for the creation of a space existing at a single point. This makes `Spaces` consistent. Calling `level` on a column field and calling `column` on a 2D horizontal field will return a field at that point. Enables #325 to be completed.

- [x] Code follows the [style guidelines](https://clima.github.io/ClimateMachine.jl/latest/DevDocs/CodeStyle/) OR N/A.
- [x] Unit tests are included OR N/A.
- [x] Code is exercised in an integration test OR N/A.
- [x] Documentation has been added/updated OR N/A.

### CODE REVIEW

File: src/Spaces/pointspace.jl [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+struct PointSpace{LG} <: AbstractPointSpace
+    local_geometry::LG
+end
+
+function PointSpace(coord::Geometry.Abstract1DPoint{FT}) where {FT}
+    CoordType = typeof(coord)
+    AIdx = Geometry.coordinate_axis(CoordType)
+    LG = Geometry.LocalGeometry{AIdx, CoordType, FT, SMatrix{1, 1, FT, 1}}
+    local_geometry = DataLayouts.DataF{LG}([Geometry.component(coord, 1) 1.0 1.0 1.0 1.0])
```
Comment (jb-mackay) [inline src/Spaces/pointspace.jl:14]: Does this seem like a fair way to initialize?

**Comment by jb-mackay** [comment]
To do:
- define `level` for `VF` to return `DataF`
- define `column` for `IJFH, IJF, IFH, IF` to return `DataF`

File: src/Spaces/pointspace.jl [hunk 2]
```diff
@@ -0,0 +1,17 @@
+abstract type AbstractPointSpace <: AbstractSpace end
+
+local_geometry_data(space::AbstractPointSpace) = space.local_geometry
+
+struct PointSpace{LG} <: AbstractPointSpace
```
Comment (simonbyrne) [inline src/Spaces/pointspace.jl:5]: Add a docstring?

File: src/Spaces/pointspace.jl (hunk 1 above)
Comment (simonbyrne) [reply src/Spaces/pointspace.jl]: I would suggest defining a constuctor:
```
PointSpace(lg::LocalGeometry)
```
and then defining other constructors in terms of that.

Comment (jb-mackay) [reply src/Spaces/pointspace.jl]: Makes sense!

Comment (jb-mackay) [reply src/Spaces/pointspace.jl]: Is there any reason not to simply make that kind of type assertion for the default constructor? e.g.
```
struct PointSpace{LG <: Geometry.LocalGeometry} <: AbstractPointSpace
    local_geometry::LG
end
```

**Comment by jb-mackay** [comment]
@simonbyrne I will get to the above comments tomorrow and will ask your advice on them as well. Please let me know if you think there are additions we should definitely make to the unit tests.

**Comment by simonbyrne** [comment]
> @simonbyrne I will get to the above comments tomorrow and will ask your advice on them as well. Please let me know if you think there are additions we should definitely make to the unit tests.

Tests seem fine. I think the next step `PointField` will require more tests

File: src/Spaces/pointspace.jl (hunk 1 above)
Comment (simonbyrne) [reply src/Spaces/pointspace.jl]: e.g. something like
```julia
function PointSpace(local_geometry::LocalGeometry)
    local_geometry_data = DataF{typeof(local_geometry(}()
    local_geometry_data[] = local_geometry
    PointSpace(local_geometry_data)
end
```

**Comment by jb-mackay** [comment]
@simonbyrne take a look and let me know if there are further tests to hit this with or if we are ready to squash & merge.

**Review by simonbyrne — APPROVED** [review]
Looks good! When you're ready please squash and merge

**Comment by jb-mackay** [comment]
bors try

**Pull request merged**